tests/bluetooth/tester/src/l2cap.c

```diff
@@ -141,41 +141,73 @@ static struct channel *get_free_channel()
 	return NULL;
 }
 
+static struct bt_l2cap_chan *get_free_channels(void)
+{
+	uint8_t i;
+	struct channel *chan;
+	struct bt_l2cap_chan *chans[CHANNELS];
+
+	for (i = 0U; i < CHANNELS; i++) {
+		if (channels[i].le.chan.state != BT_L2CAP_DISCONNECTED) {
+			continue;
+		}
+
+		chan = &channels[i];
+		chan->chan_id = i;
+
+		chans[i] = &chan->le.chan;
+	}
```

Comment:
The code below seems to assume that the number of channels returned is `CHANNELS`, but since this for loop can `continue` some of the `chans` indexes might be unassigned.